infrastructure/simulation-tool/src/operations/withdraw.ts

```diff
@@ -0,0 +1,112 @@
+import { Wallet as RollupWallet, Transaction, utils } from '@rsksmart/rif-rollup-js-sdk';
+import { sleep } from '@rsksmart/rif-rollup-js-sdk/build/utils';
+import { BigNumber } from 'ethers';
+import { getRandomBigNumber } from '../utils/number.utils';
+import config from '../utils/config.utils';
+import { getRandomElement } from './common';
+import { TransactionReceipt } from '@rsksmart/rif-rollup-js-sdk/build/types';
+
+type PreparedWithdrawal = Parameters<RollupWallet['withdrawFromSyncToRootstock']>[number] & {
+    amount: BigNumber;
+    from: RollupWallet;
+};
+
+type WithdrawResult = {
+    opL2Receipt: TransactionReceipt;
+};
+
+const prepareWithdrawal = (recipient: RollupWallet, amount?: BigNumber): PreparedWithdrawal => {
+    let [minAmount, maxAmount] = config.weiLimits.withdraw;
+    const value = amount || utils.closestPackableTransactionAmount(getRandomBigNumber(minAmount, maxAmount));
+    console.log(`Preparing withdrawal of ${value} RBTC from ${recipient.address()}.`);
+
+    return {
+        ethAddress: recipient.address(),
+        amount: value,
+        token: 'RBTC',
+        from: recipient
+    };
+};
+
+const generateWithdrawals = (numberOfWithdrawals: number, users: RollupWallet[]): PreparedWithdrawal[] => {
+    return [...Array(numberOfWithdrawals)].map((_, i) => {
+        process.stdout.write(`${i},`);
+        return prepareWithdrawal(getRandomElement(users));
+    });
+};
+
+const executeWithdrawal = async (preparedWithdrawal: PreparedWithdrawal): Promise<Transaction> => {
+    console.log(`Withdrawing ${preparedWithdrawal.amount} RBTC from ${preparedWithdrawal.from.address()}`); // commented out to reduce noise (TODO: create more sophisticated logging)
+
+    return preparedWithdrawal.from.withdrawFromSyncToRootstock(preparedWithdrawal);
+};
```

Comment:
The from is not a parameter to the method you're calling.
```suggestion
const executeWithdrawal = async ({ from, ...withdrawProps }: PreparedWithdrawal): Promise<Transaction> => {
    console.log(`Withdrawing ${withdrawProps.amount} RBTC from ${from.address()}`); // commented out to reduce noise (TODO: create more sophisticated logging)

    return from.withdrawFromSyncToRootstock(withdrawProps);
};
```